GitUI/CommandsDialogs/SettingsDialog/ShellExtension/ShellExtensionManager.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using GitCommands;
+using Microsoft.Win32;
+
+namespace GitUI.CommandsDialogs.SettingsDialog.ShellExtension
+{
+    public static class ShellExtensionManager
+    {
+        /// <summary>
+        /// Checks if necessary for registration files were found
+        /// </summary>
+        /// <returns>false only if all necessary files were found, otherwise true</returns>
+        public static bool IsNotInstalled()
+        {
+            string path32 = FindFileInBinFolders(CommonLogic.GitExtensionsShellEx32Name);
+            string path64 = FindFileInBinFolders(CommonLogic.GitExtensionsShellEx64Name);
+            return string.IsNullOrEmpty(path32) || (Environment.Is64BitOperatingSystem && string.IsNullOrEmpty(path64));
+        }
+
+        /// <summary>
+        /// Checks if shell extension is properly registered in system
+        /// </summary>
+        /// <returns>false only if all checks are positive, otherwise true</returns>
+        public static bool IsNotRegistered()
```

Comment:
```suggestion
        /// <summary>
        ///  Checks if the shell extension is properly registered in the system.
        /// </summary>
        /// <returns><see langword="true"/> if the extension is registered; otherwise <see langword="false"/>.</returns>
        public static bool CheckIfRegistered()
```